pkg/ipcache/types.go

```diff
@@ -186,208 +245,202 @@ func (s prefixInfo) sortedBySourceThenResourceID() []ipcachetypes.ResourceID {
 	})
 }
 
-func (s prefixInfo) ToLabels() labels.Labels {
-	l := labels.NewLabelsFromModel(nil)
-	for _, v := range s {
-		l.MergeLabels(v.labels)
+func (r *resourceInfo) ToLabels() labels.Labels {
+	if r.labels == nil {
+		return labels.Labels{} // code expects non-nil Labels.
 	}
-	return l
+	return r.labels
 }
 
-func (s prefixInfo) Source() source.Source {
-	src := source.Unspec
-	for _, v := range s {
-		if source.AllowOverwrite(src, v.source) {
-			src = v.source
-		}
+func (r *resourceInfo) Source() source.Source {
+	if r == nil {
+		return ""
```

Comment:
nit: Use `source.Unspec` due to the special `AllowOverwrite()` behavior for the unspec vs. empty string.
```suggestion
		return source.Unspec
```